src/app/content/utils.ts

```diff
@@ -59,3 +61,77 @@ export const scrollTocSectionIntoView = (sidebar: HTMLElement | undefined, activ
 
   sidebar.scrollTop = scrollTarget.offsetTop;
 };
+
+export const findArchiveTreeSection = (
+  book: {tree: ArchiveTree},
+  pageId: string
+): LinkedArchiveTreeSection | undefined =>
+  flattenArchiveTree(book.tree).find((section) =>
+    stripIdVersion(section.shortId) === stripIdVersion(pageId)
+    || stripIdVersion(section.id) === stripIdVersion(pageId)
+  );
+
+const getUrlParamForPageTitle = (section: LinkedArchiveTreeSection): string => {
+  const splitTitleParts = (str: string) => {
+    const match = str
+      // remove html tags from tree title
+      .replace(/<[^>]+>/g, '')
```

Comment:
Would injecting the text into a detached `<div>` be less error-prone? What if the title contains an HTML entity tag (like `&nbsp;` or `&lt;`)

Example:

```js
const div = document.createElement('div');
const strip = (xml) => {
  div.innerHTML = xml
  const stripped = div.textContent
  div.innerHTML = ''
  return stripped
}
```

Reply:
at first i was removing things like &amp; or &lt; but ended up preferring to leave them in the url, i thought it looked better that way.

&nbsp; should probably be replaced and treated like a space though.

Reply:
What about others like the matching double-quotes, math symbols, or Chinese characters? They could be removed... `webview` did that initially but then had to add them back in for SEO around the time the Polish versions of books came out.

Reply:
the replaceAccentedCharacters function replaces many wonky characters with their non-wonky equivalent for the url, anything not in that map would be stripped out by the regex